lib/jobs/ansible-job.js

```diff
@@ -31,7 +34,20 @@ function ansibleJobFactory(
         this.ansibleCmd = undefined;
         this.nodeId = this.context.target;
         this.routingKey = context.graphId;
-        this.extraVars = options.vars ? JSON.stringify(options.vars) : null;
+        this.data = undefined; 
+        this.taskKey = taskId;
+        // file for Ansible playbook to write results to:
+        this.outFileName = path.join(os.tmpdir(),'ansible-' + this.taskKey + '.out'); 
+
+        this.extraVars = '{"ansibleResultFile":"' + this.outFileName + '"}';
+        if(options.vars !== null) { 
+             this.extraVars = options.vars;
+             this.extraVars.ansibleResultFile = this.outFileName; 
+             this.extraVars = JSON.stringify(this.extraVars);
+        }
+        if (this.context.ansibleResultFile === undefined) {
+            this.context.ansibleResultFile = [];
```

Comment:
If `this.context.ansibleResultFile` is object/null/number or other non-collection types, then the line#101 `self.context.ansibleResultFile.push(data)` will fail as no `push` method is defined. So it's better to add an assertion for this.